client/rest/src/test/java/org/elasticsearch/client/HeapBufferedAsyncResponseConsumerTests.java

```diff
@@ -97,6 +102,26 @@ public void testConfiguredBufferLimit() throws Exception {
         bufferLimitTest(consumer, bufferLimit);
     }
 
+    public void testCanConfigureHeapBufferLimitFromOutsidePackage() throws ClassNotFoundException, NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        int bufferLimit = randomIntBetween(1, Integer.MAX_VALUE);
+        Class<?> aClass = Class.forName(HttpAsyncResponseConsumerFactory.HeapBufferedResponseConsumerFactory.class.getName());
+        Constructor<?> constructor = aClass.getConstructor(Integer.TYPE);
```

Comment:
I'd add a comment that you are using reflection here to make sure that the ctor is public.